src/client/Client.cc

```diff
@@ -2709,56 +2712,67 @@ void Client::handle_fs_map(const MConstRef<MFSMap>& m)
 
 void Client::handle_fs_map_user(const MConstRef<MFSMapUser>& m)
 {
+  std::scoped_lock cl(client_lock);
   fsmap_user.reset(new FSMapUser);
   *fsmap_user = m->get_fsmap();
 
   monclient->sub_got("fsmap.user", fsmap_user->get_epoch());
   signal_cond_list(waiting_for_fsmap);
 }
 
-void Client::handle_mds_map(const MConstRef<MMDSMap>& m)
+// Cancel all the commands for missing or laggy GIDs
+void Client::cancel_commands(const MDSMap* newmap)
```

Comment:
This really should be `const MDSMap&` because `newmap` cannot be null.

Reply:
Sure, I thought it is just on pointer. Fixed it now.